mod_ci/controllers.py

```diff
@@ -866,6 +867,44 @@ def show_maintenance():
     }
 
 
+@mod_ci.route('/blocked_users')
+@login_required
+@check_access_rights([Role.admin])
+@template_renderer()
+def blocked_users():
+    return {
+        'blocked_users': BlockedUsers.query.order_by(BlockedUsers.userID)
+    }
+
+
+@mod_ci.route('/blocked_users/add', method=['POST'])
+@login_required
+@check_access_rights([Role.admin])
+def add_user_to_blacklist():
+    form = AddUsersToBlacklist()
+    if form.validate_on_submit():
+        blocked_user = BlockedUsers(form.userID.data, form.comment.data)
+        if blocked_user == BlockedUsers.query.filter_by(userID=form.userID.data).first():
```

Comment:
I would just check if there's already an user with that id present (see your check for deleting). Now you compare the returned object's, which can differ on comment. If that happens, you'll get a DB-error below.